synapse/tests/test_model_dns.py

```diff
@@ -2,49 +2,90 @@
-            self.eq(t0[1].get('inet:dns:rev'), '1.2.3.4/woot.com')
-            self.eq(t0[1].get('inet:dns:rev:ipv4'), 0x01020304)
-            self.eq(t0[1].get('inet:dns:rev:fqdn'), 'woot.com')
+    def test_forms_dns_simple(self):
+        with self.getTestCore() as core:
+            with core.xact(write=True) as xact:
+                # inet:dns:a
+                node = xact.addNode('inet:dns:a', ('hehe.com', '1.2.3.4'))
+                self.eq(node.ndef[1], ('hehe.com', 0x01020304))
+                props = node.props
+                self.eq(props.get('fqdn'), 'hehe.com')
+                self.eq(props.get('ipv4'), 0x01020304)
+
+                node = xact.addNode('inet:dns:a', ('www.\u0915\u0949\u092e.com', '1.2.3.4'))
+                self.eq(node.ndef[1], ('www.xn--11b4c3d.com', 0x01020304))
+                props = node.props
+                self.eq(props.get('fqdn'), 'www.xn--11b4c3d.com')
+                self.eq(props.get('ipv4'), 0x01020304)
+
+                # inet:dns:aaaa
+                node = xact.addNode('inet:dns:aaaa', ('localhost', '::1'))
+                self.eq(node.ndef[1], ('localhost', '::1'))
+                props = node.props
+                self.eq(props.get('fqdn'), 'localhost')
+                self.eq(props.get('ipv6'), '::1')
+
+                node = xact.addNode('inet:dns:aaaa', ('hehe.com', '2001:0db8:85a3:0000:0000:8a2e:0370:7334'))
+                self.eq(node.ndef[1], ('hehe.com', '2001:db8:85a3::8a2e:370:7334'))
+                props = node.props
+                self.eq(props.get('fqdn'), 'hehe.com')
+                self.eq(props.get('ipv6'), '2001:db8:85a3::8a2e:370:7334')
+
+                # inet:dns:rev
+                node = xact.addNode('inet:dns:rev', ('1.2.3.4', 'bebe.com'))
+                self.eq(node.ndef[1], (0x01020304, 'bebe.com'))
+                props = node.props
+                self.eq(props.get('ipv4'), 0x01020304)
+                self.eq(props.get('fqdn'), 'bebe.com')
+
+                # inet:dns:rev6
+                node = xact.addNode('inet:dns:rev6', ('FF::56', 'bebe.com'))
+                self.eq(node.ndef[1], ('ff::56', 'bebe.com'))
+                props = node.props
+                self.eq(props.get('ipv6'), 'ff::56')
+                self.eq(props.get('fqdn'), 'bebe.com')
+
+                # inet:dns:ns
+                node = xact.addNode('inet:dns:ns', ('haha.com', 'ns1.haha.com'))
+                self.eq(node.ndef[1], ('haha.com', 'ns1.haha.com'))
+                props = node.props
+                self.eq(props.get('zone'), 'haha.com')
+                self.eq(props.get('ns'), 'ns1.haha.com')
+
+                # inet:dns:cname
+                node = xact.addNode('inet:dns:cname', ('HAHA.vertex.link', 'vertex.link'))
+                self.eq(node.ndef[1], ('haha.vertex.link', 'vertex.link'))
+                props = node.props
+                self.eq(props.get('fqdn'), 'haha.vertex.link')
```

Comment:
i'd encourage node.get() here (and everywhere) so others use it ( in case we wanna do any shinanigans )